### PR TITLE
polygon/bridge: Fix bridge crash at chain tip

### LINKED ISSUE #11185: astrid bridge slice bounds out of range err
Run a fresh sync from 0 using astrid standalone (--polygon.sync=true) on Amoy and got the below error coming from the bridge:
Screenshot 2024-07-06 at 18 12 04

### PR DESCRIPTION
**Pull request by shohamc1**
Fixes #11185

### CODE REVIEW

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]
please `defer it.Close()`

File: polygon/bridge/db.go [hunk 1]
```diff
@@ -165,11 +155,9 @@ func (s *MdbxStore) GetSprintLastEventID(ctx context.Context, lastID uint64, tim
 		}
 
 		eventID = event.ID
-
-		if bytes.Equal(k, kDBLast) {
-			return eventID, nil
-		}
 	}
+
+	return eventID, nil
```
Comment (antonis19) [inline polygon/bridge/db.go:160]: Shouldn't you return something to indicate the event was not found here? This the case where all the events are after the timeLimit right?

Comment (shohamc1) [reply polygon/bridge/db.go:160]: It's handled in the calling function [here](https://github.com/ledgerwatch/erigon/blob/af719e3035ed096013a60a3166ed838ca75d8086/polygon/bridge/bridge.go#L140)

Comment (antonis19) [reply polygon/bridge/db.go:160]: That seems to check something different, no ?   It is checking : `if lastDBID != 0 && lastDBID > b.lastProcessedEventID ` , but `lastBID > b.lastProcessedEventID ` is almost guaranteed by the iteration in `GetSprintLastEventID` which starts from `b.lastProcessedEventID` and going up.

I was talking about the situation where every event you iterate has  `event.Time >= timeLimit`. In that case you should have an EventNotFoundError.

**Comment by shohamc1** [comment]
That just means we have no events for that block. It should not be an error.   
  
  
  
  
 —  
Thanks
  
Shoham Chakraborty
  
  
  
  
  

  
  
>

File: polygon/bridge/db.go (hunk 1 above)
Comment (shohamc1) [reply polygon/bridge/db.go:160]: That just means we have no events for that block. It should not be an error.

Comment (antonis19) [reply polygon/bridge/db.go:160]: In that case I would suggest to return also a bool to indicate if an event was found or not, so as not to break the contract in the docstring of `GetSprintLastEventID` .

**Review by taratorio — APPROVED** [review]

**Review by antonis19 — APPROVED** [review]

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**